cypress/e2e/signUp.cy.js

```diff
@@ -1,12 +1,39 @@
 /// <reference types="cypress" />
 /// <reference types="../support" />
 
+import SignUpPageObject from '../support/pages/signUp.pageObject';
+import HomePageObject from '../support/pages/home.pageObject';
+
+const signUpPage = new SignUpPageObject();
+const homePage = new HomePageObject();
+
 describe('Sign Up page', () => {
-  before(() => {
+  let user;
+
+  beforeEach(() => {
+    cy.task('db:clear');
+    cy.task('generateUser').then(generateUser => {
+      user = generateUser;
+    });
+    signUpPage.visit();
+  });
+
+  it('should provide an ability to sign up with valid credentials', () => {
+    signUpPage.usernameField.type(user.username);
+    signUpPage.emailField.type(user.email);
+    signUpPage.passwordField.type(user.password);
+    signUpPage.signUpBtn.click();
 
+    signUpPage.assertRegistration('Your registration was successful!');
```

Comment:
Use of hardcoded strings for asserting results is not a best practice. It would be better to use constants for these strings.